l3build-arguments.lua

```diff
@@ -21,8 +21,8 @@ The development version of the bundle can be found at
 for those people who are interested.
 
 --]]
+local A = {}
```

Comment:
I'm not clear on this approach. I think you are using a local table to collect up various functions, but I'm not getting why that's superior to just adding local to those functions. I'm also wary of a very short table name being used other than in a very limited scope (e.g. inside one loop): seems a bit confusing to me.

I'd welcome comments from others!

Reply:
The name `A` simply reflects my lack of imagination. I am focusing on the principle of keeping local variables in one place. If variables must be shared, one function argument is sufficient. Avoiding globals is good practice.

Reply:
> I am focusing on the principle of keeping local variables in one place.

That's generally something I can agree with, but it doesn't seem to be consistent with this PR making *more* internal functionality externally available. 


> If variables must be shared, one function argument is sufficient. Avoiding globals is good practice.

I don't see how function arguments relate to globals so I might be missing your point here. While avoiding globals (especially global names but to a smaller degree also effective globals like ones hidden in a table) is good practice, I would generally say the same about function arguments.